tests/conference/test_conference_service.py

```diff
@@ -44,17 +44,24 @@ def test_conference_get_by_resource_id(session, conference):
     assert test_conference.conference_challenge == conference.conference_challenge
 
 
-def test_conference_get_by_resource_type():
-    raise NotImplementedError
+def test_conference_get_by_resource_type(session, conference):
+    """The service method returns a List of Conferences that match a given resource type. We'll test
+    to ensure that the first and last items returned match the desired resource type."""
+    from dispatch.conference.service import get_by_resource_type
+
+    conferences = get_by_resource_type(db_session=session, resource_type=conference.resource_type)
```

Comment:
How many objects of type Conference are you getting as a result? Looks like you're only passing one to `test_conference_get_by_resource_type()`.

Reply:
You could create a fixture that returns a list of Conferences (example [here](https://github.com/Netflix/dispatch/blob/develop/tests/conftest.py#L381)), and then add each of them to two separate incidents.

Reply:
Since it only tests whether `get_by_resource_type` returns a list of conferences matching the desired `resource_type`, this particular test really only needs one value: the contents of the `resource_type` attribute attached to the incoming `conference` object. However the fixture is a suitable workaround for testing the `get_all()` method... thanks. :smile: